### PR TITLE
fix(condo): DOMA-4014 fix duplicating roles

### PR DESCRIPTION
**Pull request by lapakota**
Ban on creating a duplicate of a role and a role with an empty name.

### CODE REVIEW

File: apps/condo/domains/contact/components/contactRoles/BaseContactRoleForm.tsx [hunk 1]
```diff
@@ -44,10 +45,19 @@ export const BaseContactRoleForm: React.FC<BaseTicketPropertyHintFormProps> = ({
...
 
+    const [existingContactRoles, setExistingContactRoles] = useState<Set<string>>()
+
     const router = useRouter()
 
     const createContactRoleAction = ContactRole.useCreate({})
     const softDeleteContactRoleAction = ContactRole.useSoftDelete()
+    const { objs: contactRoles, loading } = ContactRole.useObjects({
+        where: {
+            deletedAt: null,
```
Comment (pahaz) [inline apps/condo/domains/contact/components/contactRoles/BaseContactRoleForm.tsx:56]: Please, don’t use deletedAt it’s by default already included

File: apps/condo/domains/contact/components/contactRoles/BaseContactRoleForm.tsx [hunk 2]
```diff
@@ -44,10 +45,19 @@ export const BaseContactRoleForm: React.FC<BaseTicketPropertyHintFormProps> = ({
     const ChangesSavedMessage = intl.formatMessage({ id: 'ChangesSaved' })
     const ReadyMessage = intl.formatMessage({ id: 'Ready' })
 
+    const [existingContactRoles, setExistingContactRoles] = useState<Set<string>>()
+
     const router = useRouter()
 
     const createContactRoleAction = ContactRole.useCreate({})
     const softDeleteContactRoleAction = ContactRole.useSoftDelete()
+    const { objs: contactRoles, loading } = ContactRole.useObjects({
```
Comment (pahaz) [inline apps/condo/domains/contact/components/contactRoles/BaseContactRoleForm.tsx:54]: Please, create just a custom hook useExistingRolesNames and move All this logic to one place. Avoid to split the logic.

File: apps/condo/domains/common/hooks/useValidations.ts [hunk 3]
```diff
@@ -15,7 +14,8 @@ type ValidatorTypes = {
     lessThanValidator: (comparedValue: number, errorMessage: string) => Rule
     greaterThanValidator: (comparedValue: number, errorMessage: string, delta?: number) => Rule
     numberValidator: Rule
-    tinValidator: (country: string) => Rule
+    tinValidator: (country: string) => Rule,
```
Comment (DanilovMaxim13) [inline apps/condo/domains/common/hooks/useValidations.ts:17]: The comma is only in one line, is it really needed? :))

File: apps/condo/domains/common/hooks/useValidations.ts [hunk 4]
```diff
@@ -135,6 +136,14 @@ export const useValidations: UseValidations = (settings = {}) => {
             }
         }
 
+    const contactRoleValidator = (existingRoles: Set<string>): Rule => ({
+        validator: (_, value) => {
+            const normalizedValue = value.trim()
+            if (existingRoles.has(normalizedValue) || normalizedValue.startsWith('contact.role')) return Promise.reject(ContactRoleIsDuplicateMessage)
```
Comment (DanilovMaxim13) [inline apps/condo/domains/common/hooks/useValidations.ts:142]: very long string

File: apps/condo/domains/contact/components/contactRoles/BaseContactRoleForm.tsx (hunk 2 above)
Comment (lapakota) [reply apps/condo/domains/contact/components/contactRoles/BaseContactRoleForm.tsx]: done

File: apps/condo/domains/common/hooks/useValidations.ts (hunk 4 above)
Comment (lapakota) [reply apps/condo/domains/common/hooks/useValidations.ts]: fixed

**Pull request merged**